### PR TITLE
Added State Machine Attributes

### PR DESCRIPTION
**Pull request by lucasvdiepen**
- Added its own singleton logic
- Added state parenting functionality
- Added default state attribute

### CODE REVIEW

File: Assets/Scripts/Framework/StateMachine/State.cs [hunk 1]
```diff
@@ -8,6 +8,8 @@ namespace ProefExamen.Framework.StateMachine
     /// </summary>
     public abstract class State : MonoBehaviour
     {
+        public bool IsActive { get; private set; }
```
Comment (Joost-Robat) [inline Assets/Scripts/Framework/StateMachine/State.cs:14]: No summary for public.

File: Assets/Scripts/Framework/StateMachine/Attributes/ParentStateAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
+    public class ParentStateAttribute : Attribute
+    {
+        /// <summary>
+        /// The parent state of the state.
+        /// </summary>
+        public Type ParentState { get; }
+
+        public ParentStateAttribute(Type parentStateType) => ParentState = parentStateType;
```
Comment (Joost-Robat) [inline Assets/Scripts/Framework/StateMachine/Attributes/ParentStateAttribute.cs:16]: No summary for public.

**Review by Joost-Robat — CHANGES_REQUESTED** [review]
Some summaries missing but looking good for the rest.

File: Assets/Scripts/Framework/StateMachine/Attributes/ParentStateAttribute.cs (hunk 2 above)
Comment (lucasvdiepen) [reply Assets/Scripts/Framework/StateMachine/Attributes/ParentStateAttribute.cs:16]: This is a constructor method

**Review by Joost-Robat — APPROVED** [review]
Very nice!

**Pull request merged**